File: py/pe/pe7.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""
Find the 10001st prime.
"""

from primes import Primes

primes = Primes()

def pe7(n=10001):
    """
    >>> pe7()
    104743
    """
    return(primes[n - 1])

if __name__ == "__main__":
    import doctest
    doctest.testmod()

    try:
        while True:
            s = input('> ')
            n = int(s)
            print(pe7(n))
    except (SyntaxError, EOFError, KeyboardInterrupt, NameError):
        pass

File: cpp/pe/src/pe7.cc
#include "../include/pe7.h"
#include "../include/prime.h"

namespace pe {
  uint32_t pe7(uint32_t n)
  {
    if (n < 1) {
      return 0;
    }

    std::vector<uint32_t> ps;

    sieve(ps, n);
    if (ps.empty()) return 0;

    return ps.back();
  }
}